### PR TITLE
Added google assistant to external ressources.

### LINKED ISSUE #690: Update-external-resource: Google Assistant USAGE
Thanks to @marziman there is now a USAGE.md to be pulled over (or integrated. I'm currently not in ther loop regarding the whole `website` endeavor)

All details: https://github.com/openhab/openhab-google-assistant/issues/33

### PR DESCRIPTION
**Pull request by Confectrician**
Closes #690

Tested local and the maven build succeeded.
Assistant page was pulled and added.
Font matter is introduced in openhab/openhab-google-assistant#40

Signed-off-by: Jerome Luckenbach <github@luckenba.ch>

### CODE REVIEW

**Comment by Confectrician** [comment]
# Even if approved:

I would wait for #700 to be finished and merge directly in the new master branch then.

Wdyt @ThomDietrich and @ghys ?

**Comment by ghys** [comment]
> I would wait for #700 to be finished and merge directly in the new master branch then.

Not the master branch, but the "final" (or whatever we'll call it) :)
You can always merge and we can cherry-pick this commit afterwards.

**Comment by ghys** [comment]
I wasn't sure, now I'm sure it's not necessary to be kept in master ;)

**Comment by Confectrician** [comment]
Ok, then i would say we cherry pick it to the final branch and leave gh pages for now. :)

**Comment by Confectrician** [comment]
I think i have missed the picture.
Will have to adapt the pom with an include for that.

**Comment by Confectrician** [comment]
I have added an image execution (like its made for android and now the use doesnt get copied anymore.
Strange.

image

**Comment by Confectrician** [comment]
An hint on this @ThomDietrich or @kaikreuzer ?

I am not a maven expert, to be honest.

**Comment by ThomDietrich** [comment]
Could you update the PR with what you have?

You've added something to the middle of the file and now a step at the end of the file doesn't work anymore. Sure you did not brake the xml structure by accident?

**Comment by Confectrician** [comment]
>Sure you did not brake the xml structure by accident?

Vscode would have shown that.
I have testet it just now.

I have uploaded the corresponding commits.

**Comment by Confectrician** [comment]
I copied it 1 on 1 from the android equivalent, but maybe i have inserted a silly mistake while trying to fix it.

**Comment by Confectrician** [comment]
I have played around with the pom again and changed orders.
I have run `mvn tidy:pom` but nothing helps at all.

I give up.
Maybe one of you can figure it out.

Order of execution:
Images get copied over fine.
The readme gets copied over fine

Then there happens osmethign i cant fugure out while coming from extfolder to the final location and afterwards the readme is missing.

**Comment by ThomDietrich** [comment]
I'll have a look tomorrow :)

**Comment by ThomDietrich** [comment]
Hey,
the cause was a bit deeper than I thought and a bunch of ideas came to my head. I've solved it the easiest way for now. Have a look: https://github.com/openhab/openhab-docs/commit/7a5311d04ed0b26f80ed641b44791928b670c9a1

The main problem was the destination of your png copy process. It's not supposed to go into the temporary folder. Turns out the groovy script didn't like that. I've added tests to catch that problem next time. The script copies a full folder at this point and thanks to your png copy the folder is already available. Boom.
Obviously the script still needs some improvement. I already invested a lot of work there. Honestly I always hoped to be able to replace the cumbersome `pom.xml` file by simple bash commands and the extensive groovy script by the results of https://github.com/eclipse/smarthome/issues/4503 - which to my surprise did to this day not attract any interest...

Maybe we could discuss further simplifications in the process now possible thanks to the divide with the new website. I did not yet look into the details of all of this.

One more thing: I don't like the structure over in https://github.com/openhab/openhab-google-assistant yet. Now that we are at it, how about we move `USAGE.md` to `docs/` and the image file to `docs/images/`. As I said earlier, I'd like to establish this as the general structure for all external 
addons/sources.

Anyhow, the commit linked above contains all needed changes. Please feel free to adopt them and delete the branch ;) Best!

**Comment by Confectrician** [comment]
Yeah i thought we already agreed to a general structure and to move it in the google repo also.
I have made an example pr in the vscode repo.


About the scripts:
I would love to see some more comments there.
It is really hard to follow the stuff, when you have to dig into groovy and the regular script content parallel.

Maybe those scripts are simple for one who know what he is doing already.
That's not the case for everyone.
And in my opinion it should be easy to dig into this stuff for everyone who is willing to contribute.

**Comment by ThomDietrich** [comment]
Yes you are sure but the actual goal should be a better solution. Both the pom and the groovy file are ridiculous... I can replace the pom file by a few bash lines when time is right. The groovy file would benefit from the issue I've linked above. Anyhow, yes next time I'm touching the groovy script, I'll add comments.

Back to the task at hand. Will you create the PR? Also do you think the image will now be picked up by the new webpage?

**Comment by Confectrician** [comment]
>Will you create the PR?

There are 13 commits to compare in your branch.
I can adapt this one to the working solution next days.

>Also do you think the image will now be picked up by the new webpage?

@ghys do you have to adjust the ruby script for an additional image?

**Comment by ghys** [comment]
> @ghys do you have to adjust the ruby script for an additional image?

https://github.com/openhab/website/blob/433b743ddce3e95b27504cb9ecf6088d3b127cf6/prepare-docs.rb#L481

Yes, and the Google action should be moved out of the IO addons and into the "ecosystem" section as already done for Alexa and Mycroft:

https://github.com/openhab/website/blob/master/prepare-docs.rb#L481-L499

**Comment by Confectrician** [comment]
Ok i have added the changes from https://github.com/openhab/openhab-docs/pull/721#issuecomment-405249391 into this pr and opened up another one (#788) for the process addons change.

I have also directly prepared the pom-xml for the "new" docs structure.

process addons can be merged already imho.

This one should wait for the corresponding PR in google assistant integration.

**Comment by ghys** [comment]
> So it would be the best, if you would merge the PR when its finished and directly provide a commit for the prepare docs ruby.

Okay, but I'm not yet sure I'll have time next week.

> Since you are grabbing som IO folders anyway during the prepare script run, wouldn't it be the best to reflect the ecosystem here as a folder too in the final branch?
> I think it would be goo to clean up the prepare docs script on mid term, with providing a better adapted file structure.

Yes, you're right. The initial goal was to avoid breaking the current flow and the old docs site, but now nothing prevents altering the structure now that we don't have to worry about Jekyll anymore.
I'd also like to migrate most of the prepare-docs script logic from the website repo to here, it could be found and launched after this repo is cloned as part of the website build.

**Comment by Confectrician** [comment]
No need to hurry. This PR exists since june and has stilla dependency with the google assistant repo.

About the folders and prepare script:
Maybe we should add a Project here and split up the changes into some issues.
If we can make a roadmap and some description for this i can work on those things
and provide parallel improvements to the prepare ruby script.
I just need your help and input for collecting the needed tasks.

**Comment by Confectrician** [comment]
Closed by mistake 🤦‍♂️

File: pom.xml [hunk 1]
```diff
@@ -425,6 +425,24 @@
...
             </configuration>
           </execution>
+          <execution>
+            <id>copy-google-assistant-images</id>
+            <phase>process-resources</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${basedir}/addons/ios/google-assistant</outputDirectory>
```
Comment (ghys) [inline pom.xml]: @Confectrician I'm not sure this is right, as I understand it the goal should be to have the images under `_addons_ios/ext/google-assistant/doc` (because that's where the prepare-docs scripts expects them to be).
Same as for instance https://github.com/openhab/openhab-docs/tree/final/_addons_ios/openhabcloud

So I would move this execution step below the one copying USAGE.md to `_addons_ios/ext/google-assistant/readme.md` (to be sure the folder structure exists).

Comment (Confectrician) [reply pom.xml]: See https://github.com/openhab/openhab-docs/pull/721#issuecomment-405249391
Should be tested and working after we had a longer discussion about it.

I can test it again for sure, i f you want.

openhab cloud had no problems so far and is included for ages if i remember correctly.

Comment (Confectrician) [reply pom.xml]: Pls keep in mind that the prepare docs groovy script does some work to afterwards.
(So this could be done on another place already.)

I didn't look behind the complete process at all and just copied stuff and also trusted @ThomDietrich after his investigation of the problem.^^

The copying process is something we should simplify or at least make more read and understandable in the future anyway.

**Comment by Confectrician** [comment]
@ghys 

I did some tests (locally including the change from #788 ):
image

image
So now warnings/errors anymore

So copying throws no warnings or errors anymore.

Those are the locations with the copied content now:

image

And:
image

So indeed you seem to be right with your thoughts.
I will adapt it.

**Comment by Confectrician** [comment]
Looks way better:
image

**Comment by Confectrician** [comment]
Not completely

I would like to have:
```
_addons_ios/
 |- google-assistant/
    |- readme.md
    |- images/
       |- (*.png)
```

since this is the orga the relativ image urls in readme are expecting.

Anyway you can merge #788 already please.

**Comment by Confectrician** [comment]
Here we go:
image

I'll do a cleanup and push a last commt.
After that we are good to go.

I noticed that there seems to be a mistage in usage.md too and will file a pr for this in the google assistant repo.

**Comment by Confectrician** [comment]
:shipit:

**Comment by ghys** [comment]
🎉 
I'll do the changes to the website now while I'm at it.

**Comment by Confectrician** [comment]
"Add hot rewrite for wrong image relative path."
I will fix this in the google repo with a pr.
Should i mention/inform you, when the hot rewrite can be removed?

**Comment by ghys** [comment]
> Should i mention/inform you, when the hot rewrite can be removed?

Yes please - as a general rule. Or better, you can do a PR to remove the offending line yourself ;)

**Comment by Confectrician** [comment]
>Or better, you can do a PR to remove the offending line yourself ;)

Makes totally sense.
Sometimes the simple solutions are the one you think off at last. 😄
